modules/oe_whitelabel_starter_event/src/Plugin/Field/FieldFormatter/EventDateRangeFormatter.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_whitelabel_starter_event\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\datetime\Plugin\Field\FieldFormatter\DateTimeFormatterBase;
+use Drupal\datetime_range\DateTimeRangeTrait;
+
+/**
+ * Plugin implementation of the 'Event dates' formatter for 'daterange' fields.
+ *
+ * This formatter renders the data range as plain text, with a fully
+ * configurable date format using the PHP date syntax and separator.
+ *
+ * @FieldFormatter(
+ *   id = "event_date_range_format",
+ *   label = @Translation("Event date range format"),
+ *   field_types = {
+ *     "daterange"
+ *   }
+ * )
+ */
+class EventDateRangeFormatter extends DateTimeFormatterBase {
+
+  use DateTimeRangeTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    return [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary() {
+    return [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+
+    foreach ($items as $delta => $item) {
+      if (empty($item->start_date) && empty($item->end_date)) {
+        continue;
+      }
+
+      /** @var \Drupal\Core\Datetime\DrupalDateTime $start_date */
+      $start_date = $item->start_date;
+      /** @var \Drupal\Core\Datetime\DrupalDateTime $end_date */
+      $end_date = $item->end_date;
+
+      // Event on a single day.
+      if ($start_date->format('Y-m-d') === $end_date->format('Y-m-d')) {
```

Comment:
I like to have these comments inside the `if ()` block, if they describe the state if the condition is met.
```suggestion
      if ($start_date->format('Y-m-d') === $end_date->format('Y-m-d')) {
        // The event is on a single day.
```
or
```suggestion
      if ($start_date->format('Y-m-d') === $end_date->format('Y-m-d')) {
        // Start and end date are on the same day.
```

Reply:
See how comments outside and inside the if block use different verb forms. Outside you would say "Check if ...", inside you can make simple "is" statements.
```php
// Check if the sky is blue.
if ($sky->getColor() === 'blue') {
  // The sky is blue.
}
else {
  // The sky is not blue.
}
```